cmd/kms-rest/startcmd/start.go

```diff
@@ -188,3 +209,100 @@ func healthCheckHandler(rw http.ResponseWriter, r *http.Request) {
 		log.Errorf("healthcheck response failure, %s", err)
 	}
 }
+
+type keystoreProvider struct {
+	storageProvider storage.Provider
+	kmsCreator      provider.KMSCreator
+	crypto          crypto.Crypto
+}
+
+func (k keystoreProvider) StorageProvider() storage.Provider {
+	return k.storageProvider
+}
+
+func (k keystoreProvider) KMSCreator() provider.KMSCreator {
+	return k.kmsCreator
+}
+
+func (k keystoreProvider) Crypto() crypto.Crypto {
+	return k.crypto
+}
+
+type kmsProvider struct {
+	storageProvider ariesstorage.Provider
+	secretLock      secretlock.Service
+}
+
+func (k kmsProvider) StorageProvider() ariesstorage.Provider {
+	return k.storageProvider
+}
+
+func (k kmsProvider) SecretLock() secretlock.Service {
+	return k.secretLock
+}
+
+func createKeystoreProvider(parameters *kmsRestParameters) (provider.Provider, error) {
+	c, err := tinkcrypto.New()
+	if err != nil {
+		return nil, err
+	}
+
+	return keystoreProvider{
+		storageProvider: memstore.NewProvider(),
+		kmsCreator:      prepareKMSCreator(parameters),
+		crypto:          c,
+	}, nil
+}
+
+func prepareKMSCreator(_ *kmsRestParameters) provider.KMSCreator {
+	return func(keystoreID string) (kms.KeyManager, error) {
+		kmsStorageProv := ariesmemstorage.NewProvider()
+
+		masterKeyReader, err := prepareMasterKeyReader(kmsStorageProv)
+		if err != nil {
+			return nil, err
+		}
+
+		secretLockService, err := local.NewService(masterKeyReader, nil)
```

Comment:
`nil` argument means the master key is not encrypted, ie: master key lock is nil.

Master keys to be encrypted as well. 

See example of master key lock in the aries-framework-go unit tests [here](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/secretlock/local/local_secret_lock_test.go#L203)

Reply:
I started adding support for masterkey lock but it requires quite a lot of changes (especially logic related to getting user-specific passphrase) so decided to delegate that to the new ticket - https://github.com/trustbloc/hub-kms/issues/17